apps/web/pages/api/basenames/frame/03_getPriceAndConfirm.ts

```diff
@@ -24,6 +27,22 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     return res.status(405).json({ error: `Confirm Screen — Method (${req.method}) Not Allowed` });
   }
 
+  try {
+    const userAgent = req.headers['user-agent'] ?? 'No user agent';
+    const ip = req.headers['x-forwarded-for'] ?? req.socket.remoteAddress ?? 'No IP';
+    const deviceId = generateCustomUUID(userAgent, ip);
+    const eventName = 'selected_years';
+    const eventProperties = {
+      action: ActionType.click,
+      context: 'basenames_claim_frame',
+      componentType: ComponentType.button,
+    };
```

Comment:
it looks to me like this code could be deduplicated--what do you think?

Reply:
makes sense. will update